lib/signed-xml.js

```diff
@@ -11,33 +11,49 @@ const signatureAlgorithms = require("./signature-algorithms");
  * @type {import ("../index.d.ts").SignedXml}
  */
 class SignedXml {
-  constructor(idMode, options = {}) {
-    this.options = options;
-    this.idMode = idMode;
+  /** @param {import("../index.d.ts").SignedXmlOptions} [options={}] */
+  constructor(options = {}) {
+    const {
+      idMode,
+      idAttribute,
+      privateKey,
+      publicCert,
+      signatureAlgorithm,
+      canonicalizationAlgorithm,
+      inclusiveNamespacesPrefixList,
+      implicitTransforms,
+      keyInfoAttributes,
+      getKeyInfoContent,
+      getCertFromKeyInfo,
+    } = options;
+
+    // Options
+    this.idMode = idMode || null;
+    this.idAttributes = ["Id", "ID", "id"];
+    if (idAttribute) {
+      this.idAttributes.unshift(idAttribute);
+    }
+    this.privateKey = privateKey || null;
+    this.publicCert = publicCert || null;
+    this.signatureAlgorithm = signatureAlgorithm || "http://www.w3.org/2000/09/xmldsig#rsa-sha1";
```

Comment:
This isn't appropriate for this PR, but I'm wondering if we should upgrade this to be more secure by default.

Reply:
I have no strong opinion regarding this.